### PR TITLE
Vitamins_included keymap vitavim

### PR DESCRIPTION
**Pull request by ddylewsky**
Added new keymap for the keyboard vitamins included. 

## Description

Added a new keyboard keymap and config for the vitamins included keyboard. In the config I added RGBLIGHT_SPLIT which syncs the under glow rgb's for both sides. Without this any rgb key only effects one side. Added my own custom key map that includes some vim navigation layers, and a num pad layer among others.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/vitamins_included/keymaps/vitavim/keymap.c [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      break;
+    case CODE:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_CODE);
+        }
+      return false;
+      break;
+  }
+  return true;
+}
```
Comment (zvecr) [inline keyboards/vitamins_included/keymaps/vitavim/keymap.c:153]:
```suggestion
```

File: keyboards/vitamins_included/keymaps/vitavim/keymap.c [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+#define _CODE 4
+
+enum custom_keycodes {
+  ALPHAS = SAFE_RANGE,
+  MODS,
+  VIM,
+  NUMPAD,
+  CODE
+};
+
```
Comment (zvecr) [inline keyboards/vitamins_included/keymaps/vitavim/keymap.c:22]:
```suggestion

```

File: keyboards/vitamins_included/keymaps/vitavim/keymap.c [hunk 3]
```diff
@@ -0,0 +1,153 @@
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
+
```
Comment (zvecr) [inline keyboards/vitamins_included/keymaps/vitavim/keymap.c:4]:
```suggestion

```

File: keyboards/vitamins_included/keymaps/vitavim/config.h [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+// Defaults:
+// Rev1: EE_HANDS
+// Rev2: SPLIT_HAND_PIN B4
+
+// You can override the defaults (rev1 doesn't support SPLIT_HAND_PIN)
+//#define EE_HANDS
+//#define MASTER_LEFT
+//#define MASTER_RIGHT
+
+#define RGBLIGHT_SPLIT
```
Comment (zvecr) [inline keyboards/vitamins_included/keymaps/vitavim/config.h:36]: Is this a rev2 board?

Comment (ddylewsky) [reply keyboards/vitamins_included/keymaps/vitavim/config.h:36]: Yes, just got it from NovelKeys_

Comment (Duckle29) [inline keyboards/vitamins_included/keymaps/vitavim/config.h:36]:
```suggestion
```

Thanks for getting the board, and for putting some time into this, and reminding me I had to get RGBSPLIT implemented :P

https://github.com/qmk/qmk_firmware/pull/8871 < I've opened this PR which should enable RGBSPLIT for rev2 per default, as well as a few other things.

Comment (ddylewsky) [reply keyboards/vitamins_included/keymaps/vitavim/config.h:36]: No problem, happy to contribute. Loving the board and with the case from keebio with tenting it is amazing. Thanks!

File: keyboards/vitamins_included/keymaps/vitavim/keymap.c [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+ * |LSPO    |   z  |  x  |   c  |    v    |  b   |   n  |   m   |  ,<  |  .>  |   up   |  RSPC  |
+ * |--------+------+-----+------+---------+------+------+-------+------+------+--------+--------|
+ * | ctl    |  GUI | alt |MO(1) |lt(3,del)|Space |Space |bckspc |MO(4) | left |  down  |  right |
+ * `--------------------------------------------------------------------------------------------'
+ */
+[_ALPHAS] = LAYOUT_ortho_4x12( \
+  KC_TAB,         KC_Q,    KC_W,    KC_E,  KC_R,         KC_T,   KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,          KC_INS, \
+  LCTL_T(KC_ESC), KC_A,    KC_S,    KC_D,  KC_F,         KC_G,   KC_H,   KC_J,    KC_K,    KC_L,    LT(2,KC_SCLN), RSFT_T(KC_ENT), \
+  KC_LSPO,        KC_Z,    KC_X,    KC_C,  KC_V,         KC_B,   KC_N,   KC_M,    KC_COMM, KC_DOT,  KC_UP,         KC_RSPC, \
+  KC_LCTL,        KC_LGUI, KC_LALT, MO(1), LT(3,KC_DEL), KC_SPC, KC_SPC, KC_BSPC, MO(4),   KC_LEFT, KC_DOWN,       KC_RGHT \
```
Comment (fauxpark) [inline keyboards/vitamins_included/keymaps/vitavim/keymap.c:28]:
```suggestion
[_ALPHAS] = LAYOUT_ortho_4x12(
  KC_TAB,         KC_Q,    KC_W,    KC_E,  KC_R,         KC_T,   KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,          KC_INS,
  LCTL_T(KC_ESC), KC_A,    KC_S,    KC_D,  KC_F,         KC_G,   KC_H,   KC_J,    KC_K,    KC_L,    LT(2,KC_SCLN), RSFT_T(KC_ENT),
  KC_LSPO,        KC_Z,    KC_X,    KC_C,  KC_V,         KC_B,   KC_N,   KC_M,    KC_COMM, KC_DOT,  KC_UP,         KC_RSPC,
  KC_LCTL,        KC_LGUI, KC_LALT, MO(1), LT(3,KC_DEL), KC_SPC, KC_SPC, KC_BSPC, MO(4),   KC_LEFT, KC_DOWN,       KC_RGHT
```

File: keyboards/vitamins_included/keymaps/vitavim/keymap.c [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+ * | LS/(  |  z   |   x  |  [{  |  ]   |   b  |  -_  |  =+  |  /?  |  \|  | PgUp | RS/) |
+ * |-------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Ctrl  |      |      |      |      | spc  |  _   |bckspc|      | Home |PgDwn | End  |
+ * `------------------------------------------------------------------------------------'
+ */
+[_MODS] =  LAYOUT_ortho_4x12( \
+    KC_GRV,          KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_PSCR, \
+    MEH(KC_F7),      KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_QUOT, KC_DQUO, KC_BSLS, \
+    LSFT_T(KC_LCBR), KC_Z,    KC_X,    KC_LBRC, KC_RBRC, KC_B,    KC_MINS, KC_EQL,  KC_SLSH, KC_BSLS, KC_PGUP, RSFT_T(KC_RCBR), \
+    KC_LCTL,         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_SPC,  KC_UNDS, KC_BSPC, KC_NO,   KC_HOME, KC_PGDN, KC_END \
```
Comment (fauxpark) [inline keyboards/vitamins_included/keymaps/vitavim/keymap.c:46]:
```suggestion
[_MODS] =  LAYOUT_ortho_4x12(
    KC_GRV,          KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_PSCR,
    MEH(KC_F7),      KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_QUOT, KC_DQUO, KC_BSLS,
    LSFT_T(KC_LCBR), KC_Z,    KC_X,    KC_LBRC, KC_RBRC, KC_B,    KC_MINS, KC_EQL,  KC_SLSH, KC_BSLS, KC_PGUP, RSFT_T(KC_RCBR),
    KC_LCTL,         KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_SPC,  KC_UNDS, KC_BSPC, KC_NO,   KC_HOME, KC_PGDN, KC_END
```

File: keyboards/vitamins_included/keymaps/vitavim/keymap.c [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+ * | sft    |  ;:  |  q  |  ^  |   $    |  x  |alt(=)|ctl(+)|ctl(-)|      |        |      |
+ * |--------+------+-----+-----+--------+-----+------+------+------+------+--------+------|
+ * | Ctl    |  GUI | alt | sft |sft(ctl)| spc |  del | bspc |      |      |        |      |
+ * `--------------------------------------------------------------------------------------'
+ */
+[_VIM] =  LAYOUT_ortho_4x12( \
+    KC_TAB,         KC_QUOT, KC_B,    KC_W,    KC_E,          KC_D,   KC_Y,         KC_VOLD,       KC_VOLU,       TO(0),   TG(2),         RESET, \
+    LCTL_T(KC_ESC), KC_HOME, KC_HOME, KC_PGUP, KC_PGDN,       KC_END, KC_LEFT,      KC_DOWN,       KC_UP,         KC_RGHT, LT(2,KC_SCLN), KC_ENT, \
+    KC_LSFT,        KC_SCLN, KC_Q,    KC_CIRC, KC_DLR,        KC_X,   RALT(KC_EQL), RCTL(KC_PLUS), RCTL(KC_PMNS), KC_NO,   KC_NO,         KC_TRNS, \
+    KC_LCTL,        KC_LGUI, KC_LALT, KC_LSFT, LSFT(KC_LCTL), KC_SPC, KC_DEL,       KC_BSPC,       KC_NO,         KC_NO,   KC_NO,         KC_TRNS \
```
Comment (fauxpark) [inline keyboards/vitamins_included/keymaps/vitavim/keymap.c:65]:
```suggestion
[_VIM] =  LAYOUT_ortho_4x12(
    KC_TAB,         KC_QUOT, KC_B,    KC_W,    KC_E,          KC_D,   KC_Y,         KC_VOLD,       KC_VOLU,       TO(0),   TG(2),         RESET,
    LCTL_T(KC_ESC), KC_HOME, KC_HOME, KC_PGUP, KC_PGDN,       KC_END, KC_LEFT,      KC_DOWN,       KC_UP,         KC_RGHT, LT(2,KC_SCLN), KC_ENT,
    KC_LSFT,        KC_SCLN, KC_Q,    KC_CIRC, KC_DLR,        KC_X,   RALT(KC_EQL), RCTL(KC_PLUS), RCTL(KC_PMNS), KC_NO,   KC_NO,         KC_TRNS,
    KC_LCTL,        KC_LGUI, KC_LALT, KC_LSFT, LSFT(KC_LCTL), KC_SPC, KC_DEL,       KC_BSPC,       KC_NO,         KC_NO,   KC_NO,         KC_TRNS
```

File: keyboards/vitamins_included/keymaps/vitavim/keymap.c [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+ * | sft( |      |      |      |      |      |      |   1  |  2   |  3   |      |kc_sft()|
+ * |------+------+------+------+------+------+------+------+------+------+------+--------|
+ * |      |      |      |      |      |Space |Space |   0  |      |      |      |        |
+ * `-------------------------------------------------------------------------------------'
+ */
+[_NUMPAD] =  LAYOUT_ortho_4x12( \
+    KC_LBRC, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_LPRN, KC_7, KC_8,  KC_9,  KC_RPRN, KC_RBRC, \
+    KC_LCBR, KC_PSLS, KC_PAST, KC_PMNS, KC_PPLS, KC_NO,   KC_EQL,  KC_4, KC_5,  KC_6,  KC_SCLN, KC_RCBR, \
+    KC_LSPO, KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_TRNS, KC_1, KC_2,  KC_3,  KC_PEQL, KC_RSPC, \
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_SPC,  KC_0, KC_NO, KC_NO, KC_NO,   KC_NO \
```
Comment (fauxpark) [inline keyboards/vitamins_included/keymaps/vitavim/keymap.c:83]:
```suggestion
[_NUMPAD] =  LAYOUT_ortho_4x12(
    KC_LBRC, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_LPRN, KC_7, KC_8,  KC_9,  KC_RPRN, KC_RBRC,
    KC_LCBR, KC_PSLS, KC_PAST, KC_PMNS, KC_PPLS, KC_NO,   KC_EQL,  KC_4, KC_5,  KC_6,  KC_SCLN, KC_RCBR,
    KC_LSPO, KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_TRNS, KC_1, KC_2,  KC_3,  KC_PEQL, KC_RSPC,
    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_SPC,  KC_0, KC_NO, KC_NO, KC_NO,   KC_NO
```

File: keyboards/vitamins_included/keymaps/vitavim/keymap.c [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+ * |    | F7  | F8  | F9 | F10 | F11 |   ~  |  =   |  \   |RGBTOG|RGBHUI|RGBHUD|
+ * |----+-----+-----+----+-----+-----+------+------+------+------+------+------|
+ * |    |     |     |    |     |     |      |      |      |RGBVAD|RBGVAI|RGBMOD|
+ * `---------------------------------------------------------------------------'
+ */
+[_CODE] =  LAYOUT_ortho_4x12( \
+    KC_GRV,  KC_F11,  KC_F12,  KC_3,    KC_4,    KC_5,    KC_6,    KC_COLN, KC_MINS, KC_GT,   KC_0,    KC_DEL, \
+    KC_TRNS, KC_PSLS, KC_PAST, KC_LCBR, KC_LBRC, KC_LPRN, KC_RPRN, KC_RBRC, KC_RCBR, KC_QUOT, KC_RBRC, KC_BSLS, \
+    KC_TRNS, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_TILD, KC_EQL,  KC_NUBS, RGB_TOG, RGB_HUI, RGB_HUD, \
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_VAD, RGB_VAI, RGB_MOD \
```
Comment (fauxpark) [inline keyboards/vitamins_included/keymaps/vitavim/keymap.c:101]:
```suggestion
[_CODE] =  LAYOUT_ortho_4x12(
    KC_GRV,  KC_F11,  KC_F12,  KC_3,    KC_4,    KC_5,    KC_6,    KC_COLN, KC_MINS, KC_GT,   KC_0,    KC_DEL,
    KC_TRNS, KC_PSLS, KC_PAST, KC_LCBR, KC_LBRC, KC_LPRN, KC_RPRN, KC_RBRC, KC_RCBR, KC_QUOT, KC_RBRC, KC_BSLS,
    KC_TRNS, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_TILD, KC_EQL,  KC_NUBS, RGB_TOG, RGB_HUI, RGB_HUD,
    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_VAD, RGB_VAI, RGB_MOD
```

File: keyboards/vitamins_included/keymaps/vitavim/keymap.c [hunk 10]
```diff
@@ -0,0 +1,104 @@
+#include QMK_KEYBOARD_H
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _ALPHAS 0
+#define _MODS 1
+#define _VIM 2
+#define _NUMPAD 3
+#define _CODE 4
```
Comment (drashna) [inline keyboards/vitamins_included/keymaps/vitavim/keymap.c:10]: We recommend using enums for the layer definitions, as they're easier to maintain, long term. 

```suggestion
enum layers {
    _ALPHAS,
    _MODS,
    _VIM,
    _NUMPAD,
    _CODE,
};
```

**Comment by noroadsleft** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Pull request merged**